great_expectations/core/result_format.py

```diff
@@ -0,0 +1,14 @@
+import enum
+
+from typing_extensions import TypedDict
+
+
+class ResultFormat(str, enum.Enum):
+    BOOLEAN_ONLY = "BOOLEAN_ONLY"
+    BASIC = "BASIC"
+    COMPLETE = "COMPLETE"
+    SUMMARY = "SUMMARY"
+
+
+class ResultFormatDict(TypedDict):
+    result_format: ResultFormat
```

Comment:
Don't fully know the shape of this but this will provide some basic structure until we refactor/rework result format